internal/repository/mongo/employee/employee.go

```diff
@@ -0,0 +1,152 @@
+package employee
+
+import (
+	"context"
+	"errors"
+	"github.com/dilyara4949/employees-api/internal/domain"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"log"
+	"time"
+)
+
+type employeeRepository struct {
+	employeeCollection *mongo.Collection
+	positionCollection *mongo.Collection
+}
+
+func NewEmployeesRepository(db *mongo.Database, emp, pos string) domain.EmployeesRepository {
+	return &employeeRepository{
+		employeeCollection: db.Collection(emp),
+		positionCollection: db.Collection(pos),
+	}
+}
+
+var (
+	ErrEmployeeNotFound = errors.New("employee not found")
+	ErrNothingChanged   = errors.New("nothing changed")
+)
+
+func (e *employeeRepository) Create(ctx context.Context, employee domain.Employee) (*domain.Employee, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	position := domain.Position{}
+	if err := e.positionCollection.FindOne(ctx, bson.M{"id": employee.PositionID}).Decode(&position); err != nil {
+		return nil, err
+	}
+
+	_, err := e.employeeCollection.InsertOne(ctx, bson.M{
```

Comment:
NIT: you have `bson` tags on your `employee` structure, so you can serialize it using them.

Reply:
while creating entity, i'll need created_at extra field since domain entity don't have it, i would add new entity with those extra fields. I think it is ok  to use bson object as  it will be needed at create case only.

Reply:
Yeah, it's just a note for future, no need to change for now. However you need to be aware, that you need to change these type of explicit conversions too, when schema changes.

Reply:
no action required